### ISSUE TITLE
About anchors

### ISSUE THREAD

**Issue by Egorundel** [ultralytics/ultralytics#18653]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I have trained a neural network YOLOv11, but it does not detect large objects in the vicinity well. For example, if a car is parked nearby and only its hood and windshield are visible, then the car is not detected. 

However, YOLOv4 detects such a machine. I think the anchors are the problem.
Can you help me figure it out? Where can I set up YOLO layers or anchors in YOLOv11?

Here is an example of a YOLOv4 configuration that works well with nearby objects.:
https://github.com/AlexeyAB/darknet/blob/9ade741db91fd3d796d2abb0c9889b10943ea28a/cfg/yolov4.cfg

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Egorundel, thank you for your interest in Ultralytics 🚀! We appreciate your detailed question and example regarding YOLOv11 and anchor configurations.

We recommend visiting the [Docs](https://docs.ultralytics.com) where you'll find helpful resources on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. These guides might give you insights into custom configurations such as adjusting model components, though YOLOv11 adopts a more modern architecture, reducing the need for traditional anchor fine-tuning.

If this is specifically a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can help debug it effectively. Ensure this includes your training script, dataset details, or any configurations you're using.

If this is a training or architecture ❓ Question, please provide additional context, such as YOLOv11 training configurations, any customizations, and specific dataset examples. You can also verify you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to help improve model performance.

We encourage you to explore the Ultralytics community to discuss these types of topics further:
- Real-time help and discussions: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth explorations: [Discourse](https://community.ultralytics.com)
- Sharing knowledge and experiences: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you're using the latest version of the `ultralytics` package. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify if improvements are already available:

```bash
pip install -U ultralytics
```

## Environments

YOLOv11 can be run in verified environments such as:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

🤖 This is an automated response, but an Ultralytics engineer will engage with you shortly to assist further!

**Comment by Y-T-G** [comment]
You can try P6 model.

https://github.com/ultralytics/ultralytics/issues/16895#issuecomment-2413016050

**Comment by Egorundel** [comment]
> You can try P6 model.
> 
> [#16895 (comment)](https://github.com/ultralytics/ultralytics/issues/16895#issuecomment-2413016050)

Thanks, I'll try. Do you have a P6 model configuration for YOLOv11?

**Comment by Egorundel** [comment]
> You can check this
> 
> #16558

Thanks! I think I found it, if I'm not mistaken.

https://github.com/WangQvQ/ultralytics/blob/e43a6665db5422cf75d349e851a93a07c66852fc/ultralytics/cfg/models/11/yolo11-p6.yaml

yolo11-p6.yaml:
```yaml
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLO11 object detection model with P3-P6 outputs. For Usage examples see https://docs.ultralytics.com/tasks/detect

# Parameters
nc: 80 # number of classes
scales: # model compound scaling constants, i.e. 'model=yolo11n.yaml' will call yolo11.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.50, 0.25, 1024]
  s: [0.50, 0.50, 1024]
  m: [0.50, 1.00, 512]
  l: [1.00, 1.00, 512]
  x: [1.00, 1.50, 512]

# YOLO11-P6 backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]] # 1-P2/4
  - [-1, 2, C3k2, [128, False, 0.25]]
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8
  - [-1, 2, C3k2, [256, False, 0.25]]
  - [-1, 1, Conv, [512, 3, 2]] # 5-P4/16
  - [-1, 2, C3k2, [512, True]]
  - [-1, 1, Conv, [768, 3, 2]] # 7-P5/32
  - [-1, 2, C3k2, [768, True]]
  - [-1, 1, Conv, [1024, 3, 2]] # 9-P6/64
  - [-1, 2, C3k2, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 11
  - [-1, 2, C2PSA, [1024]] # 12

# YOLO11-P6 head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 8], 1, Concat, [1]] # cat backbone P5
  - [-1, 2, C3k2, [512, False]] # 15

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 2, C3k2, [512, False]] # 18

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 2, C3k2, [256, False]] # 21 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 18], 1, Concat, [1]] # cat head P4
  - [-1, 2, C3k2, [512, False]] # 24 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 15], 1, Concat, [1]] # cat head P5
  - [-1, 2, C3k2, [768, True]] # 27 (P5/32-large)

  - [-1, 1, Conv, [768, 3, 2]]
  - [[-1, 12], 1, Concat, [1]] # cat head P6
  - [-1, 2, C3k2, [1024, True]] # 30 (P6/64-xlarge)

  - [[21, 24, 27, 30], 1, Detect, [nc]] # Detect(P3, P4, P5, P6)
  ```

**Comment by Egorundel** [comment]
@Y-T-G 

And then how can I use this P6 configuration for training? 

That's how to write in the file `train.py`?
```python
model = YOLO("yolo11-p6.yaml").load("yolo11m.pt")
```

**Comment by Y-T-G** [comment]
`YOLO("yolo11m-p6.yaml").load("yolo11n.pt")`

The filename you saive it to doesn't need to have `m`. It will automatically parse it.

**Comment by Egorundel** [comment]
@Y-T-G Thank you very much! 

One last question. What do P2, P3, ... , P6 even mean?

**Comment by Y-T-G** [comment]
Feature Pyramid Levels. A higher number detects larger objects.

**Comment by Egorundel** [comment]
@Y-T-G Thank you for your help! 

The training seems to be underway :)
